clang-tools-extra/docs/clang-tidy/checks/modernize/use-integer-sign-comparison.rst

```diff
@@ -0,0 +1,43 @@
+.. title:: clang-tidy - modernize-use-integer-sign-comparison
+
+modernize-use-integer-sign-comparison
+=====================================
+
+Replace comparisons between signed and unsigned integers with their safe
+``std::cmp_*`` alternative.
+
+Examples of fixes created by the check:
+
+.. code-block:: c++
+
+    uint func(int bla)
+    {
+        uint result;
+        if (result == bla)
+            return 0;
+
+        return 1;
+    }
```

Comment:
Using 
```c++
uint func(int a, uint b) {
  return a == b;
}
```

instead would mean that the example is not using an uninitialized variable or unnecessary `if`.

+- on the `uint` instead of something that exists